src/file_change_handler.ts

```diff
@@ -71,4 +80,12 @@ export class FileChangeHandler {
 			id: "" // TODO: Fix this, should be optional!
 		}
 	}
+
+	private isDartLike(document: vscode.TextDocument): boolean {
```

Comment:
I think it'd be useful to filter - to just create overlays for things we know are dart/dartlike files. +1 to improving this list to what's in that email: `'*.dart', '*.html', '*.htm', '.analysis_options', and 'analysis_options.yaml'`.